src/Nodes/DeviceNode.ts

```diff
@@ -19,10 +19,10 @@ export class DeviceNode implements INode {
         return this.deviceItem;
     }
 
-    public async getChildren(): Promise<INode[]> {
+    public async getChildren(context: vscode.ExtensionContext, iotHubConnectionString: string): Promise<INode[]> {
         let nodeList: INode[] = [];
         nodeList.push(new ModuleLabelNode(this));
-        if (this.deviceItem.contextValue === "device") {
+        if (this.deviceItem.contextValue === "device" && iotHubConnectionString.indexOf("azure-devices.cn;") < 0) {
```

Comment:
is indexOf case insensitive?

Reply:
I tested the connection can accept capitalize url, and I updated the code to handle this